File: python/insertion_sort.py
def insertion_sort(N, A):
    for i in range (N):
        v = A[i]
        j = i - 1
        while j >= 0 and A[j] > v:
            A[j+1] = A[j]
            j = j -1
        A[j+1] = v

if __name__ == '__main__':
    N = input()
    A = input().split()

    insertion_sort(N, A)

File: ruby/insertion_sort.rb
def insertion_sort(n, a)
  for i in 0..n-1
    v = a[i]
    j = i - 1
    while j >= 0 && a[j] > v
      a[j+1] = a[j]
      j = j - 1
    end
    a[j+1] = v
    print a.join(" ") + "\n"
  end
end

if __FILE__ == $0
  n = gets.to_i
  a = gets.split(" ").map(&:to_i)
  p a
  insertion_sort(n, a)
end
